### PR TITLE
[added] property disabled on MenuItem

### LINKED ISSUE #246: Disable MenuItem
I have hacked my MenuItem code to include a `propTypes { disabled: React.PropTypes.bool }`

In recent browsers, (Chrome, IE11) the `transferPropsTo` in the render method produces <li disabled> which renders them as disabled with a "no-entry" icon, but earlier browsers don't respond to the disabled attribute. I have therefore added disabled to the class map:

    var classes = {
        'dropdown-header': this.props.header,
        'divider': this.props.divider,
        'disabled': this.props.disabled
      };

and I have CSS:

```
.disabled {
	color: lightgray;
	cursor: not-allowed;
}
```

But in general, it might be better to allow arbitrary classes to be passed down to the `<li>` - what do you think?

### PR DESCRIPTION
**Pull request by dozoisch**
The test will currently fail, but I'll rerun them when https://github.com/react-bootstrap/react-bootstrap/pull/604 gets merged.

closes #246

### CODE REVIEW

File: docs/src/ComponentsPage.js [hunk 1]
```diff
@@ -574,6 +574,24 @@ const ComponentsPage = React.createClass({
                   <code>getValue()</code> will not work when used this way.</p>
                   <ReactPlayground codeText={Samples.InputWrapper} />
                 </div>
+
+                {/* Menuitem */}
+                <div className='bs-docs-section'>
+                  <h1 id='menu-item' className='page-header'>MenuItem</h1>
```
Comment (mtscout6) [inline docs/src/ComponentsPage.js]: For consistency this title should probably follow the same format as [List group](http://localhost:4000/components.html#listgroup). Note an English friendly spaced variant with `small` component names to its side.

File: docs/src/ComponentsPage.js [hunk 2]
```diff
@@ -574,6 +574,24 @@ const ComponentsPage = React.createClass({
...
+                      <li><code>header</code>: To add a header label to sections</li>
+                      <li><code>divider</code>: Adds an horizontal divider between sections</li>
+                      <li><code>disabled</code>: shows the item as disabled, and prevents the onclick</li>
+                      <li><code>eventKey</code>: passed to the callback</li>
+                      <li><code>onSelect</code>: a callback that is called when the user clicks the item.</li>
+                    </ul>
+                  <p>The callback is called with the following arguments: <code>eventKey</code>, <code>href</code> and <code>target</code></p>
+                  </p>
+                  <ReactPlayground codeText={Samples.MenuItem} />
+                </div>
```
Comment (mtscout6) [inline docs/src/ComponentsPage.js]: I wonder about positioning of this section. Should we put this next to either the dropdown buttons or nav sections?

Comment (dozoisch) [inline docs/src/ComponentsPage.js]: Yeah I was wondering where to put it. I guess we should put it next to dropdown like bootstrap does.

File: docs/src/ComponentsPage.js (hunk 1 above)
Comment (dozoisch) [inline docs/src/ComponentsPage.js]: Alright I will fix it.

File: docs/src/ComponentsPage.js (hunk 2 above)
Comment (mtscout6) [inline docs/src/ComponentsPage.js]: :+1:

**Comment by dozoisch** [comment]
alright Ill try to address the issue when I get some time today

**Comment by dozoisch** [comment]
@mtscout6  with the revisited MenuItem is this still needed? Or should I implemented in the dropdown-reviited branch?

**Comment by mtscout6** [comment]
I think this can come into master now, You can add it to the dropdown-revisited branch as well if you like. I've been rebasing and force pushing that branch regularly to keep it up to date. If you don't push it there then I'll be sure to account for it.

**Comment by mtscout6** [comment]
I added a check to make sure I account for it.

**Comment by dozoisch** [comment]
Alright, I'll clean it tonight so we can merge it

**Comment by dozoisch** [comment]
@mtscout6 alright, it's updated now!

File: docs/src/ComponentsPage.js [hunk 3]
```diff
@@ -170,6 +170,25 @@ const ComponentsPage = React.createClass({
                   <ReactPlayground codeText={Samples.SplitButtonRight} />
                 </div>
 
+                {/* Menu Item */}
+                <div className='bs-docs-section'>
+                  <h1 id='menu-item' className='page-header'>Menu Item <small> MenudItem</small></h1>
+                  <p>This is a component used in other components (see <a href="buttons">Buttons</a>, <a href="#navbars">Navbar</a>).</p>
```
Comment (AlexKVal) [inline docs/src/ComponentsPage.js]: `#navbars">Navbar<a`
Navbar => Navbars

**Comment by AlexKVal** [comment]
From the English language point of view
`active` vs `disabled` :smile: two properties.

And both are legitimate from `lib` point of view.

I'm not sure does it need a documenting note about them.
In theory `Bootstrap` users have to know about this `active` and `disabled` as classes
and therefore they should not have questions about it.

**Comment by AlexKVal** [comment]
`This is a component used in other components (see Buttons, Navbar).`
It is added into `Buttons` submenu.
Should we add it into `Navbars`-submenu too ?

**Comment by mtscout6** [comment]
If you add a prop for active as well, then we should add prop type validations to ensure that only one of them is used at a time. I'm in favor of the idea, though not required in this PR as it could be followed up in another.

**Comment by dozoisch** [comment]
I didn't add the active prop, it was already there!

**Comment by taion** [comment]
@mtscout6

We don't have that validation for e.g. `Button`: https://github.com/react-bootstrap/react-bootstrap/blob/1b98730d16a128566ad48159ecab419720988992/src/Button.js#L9-L10

I don't think we need to enforce that invariant - it's potentially not logically meaningless, and it doesn't break anything internally for us (I think?).

**Comment by mtscout6** [comment]
Ah, I guess they are semantically different anyways my bad.

**Comment by dozoisch** [comment]
Alright, I can correct the Navbar to Navbars tonight if needed, but will have to wait the whole day!

**Comment by AlexKVal** [comment]
It's possible to click on `disabled` and it gains focus.
I've looked at disabled `Button`, it doesn't allow focusing by these css:
```css
cursor: not-allowed;
pointer-events: none;
```
Though I'm not sure does the component need this.

screen shot 2015-05-28 at 7 59 45 pm

**Comment by dozoisch** [comment]
@mtscout6 @AlexKVal  alright navbar => navbars has been corrected. Sorry for the delay

**Comment by AlexKVal** [comment]
It is added into Buttons submenu.
Should we add it into Navbars-submenu too ?

If we should it and you just have no time for it, then I can add it after merge.
Code LGTM.

File: docs/examples/MenuItem.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  alert('Alert from menu item.\neventKey: "' + eventKey + '"\nhref: "' + href + '"');
+}
+
+function preventDefault() {}
+
+const MenuItems = (
+  <div className="clearfix">
+    <ul className="dropdown-menu open">
+      <MenuItem header>Header</MenuItem>
+      <MenuItem onSelect={preventDefault}>link</MenuItem>
```
Comment (mtscout6) [inline docs/examples/MenuItem.js]: I'm not a fan of this, though I'm sure I'll have this cleaned up in the dropdown revisited branch so I guess we can live with it in the time being.

File: test/MenuItemSpec.js [hunk 5]
```diff
@@ -107,4 +107,13 @@ describe('MenuItem', function () {
...
   });
+
+  it('Should be `disabled` link', function () {
+    let instance = ReactTestUtils.renderIntoDocument(
+      <MenuItem disabled>
+        Title
+      </MenuItem>
+    );
+    assert.ok(instance.getDOMNode().className.match(/\bdisabled\b/));
+  });
```
Comment (mtscout6) [inline test/MenuItemSpec.js]: Perhaps adding a test for the click event to ensure that `e.preventDefault()` is called when disabled.

File: docs/examples/MenuItem.js (hunk 4 above)
Comment (mtscout6) [inline docs/examples/MenuItem.js]: I have this cleaned up in the dropdown-revisited branch now.

File: test/MenuItemSpec.js (hunk 5 above)
Comment (mtscout6) [inline test/MenuItemSpec.js]: This is covered in the dropdown-revisited branch now.

**Comment by mtscout6** [comment]
LGTM

**Comment by dozoisch** [comment]
@mtscout6 Thx for cleaning these up in dropdown-revisited... I'm completely struggling with time these days :/

**Comment by mtscout6** [comment]
No worries friend, that's why the Maintaining guide says do what you can when you can. We understand when other priorities take precedence.

**Pull request merged**